logp/logger.go

```diff
@@ -224,6 +224,11 @@ func (l *Logger) Sync() error {
 	return l.logger.Sync()
 }
 
+// Core returns the backend zapcore.Core for the logger.
+func (l *Logger) Core() zapcore.Core {
```

Comment:
I wonder if it would be better to define a type alias like it's done with the `zap.Option`:

https://github.com/elastic/elastic-agent-libs/blob/89215adc2946dcd3cfc873fb954c51eb7fb8fd91/logp/logger.go#L27-L28

I don't know how far we want to go exposing the `zap` types in our library. I see that the `Logger` struct just has `zap` types directly:

https://github.com/elastic/elastic-agent-libs/blob/89215adc2946dcd3cfc873fb954c51eb7fb8fd91/logp/logger.go#L30-L34

Reply:
It would rather an abstraction should we decide to switch from zap to something else. I guess that's the point. But I'm okay with this.